### ISSUE TITLE
Can't overload async/await methods with NUnit Test Adapter

### ISSUE THREAD

**Issue by ErikCL** [nunit/nunit-vs-adapter#87]
I'm getting a `System.Reflection.AmbiguousMatchException` when **at least one** of the methods are async **and the method is overloaded** when the NUnit tests are run using the NUnit Test Adapter 2.0.0.0 in Visual Studio. I'm able to run the tests without problem when I use the ReSharper unit test runner. I also posted this on [StackOverflow](http://stackoverflow.com/q/33620614/5190842)

````c#
    [TestFixture]
    public class SimpleReproAttempt
    {
        [Test]
        [TestCase("I'm valid")]
        [TestCase("So am I!")]
        public async Task Foo(string resource)
        {
            Assert.IsNotNull(resource);

            await Task.Delay(new TimeSpan(0, 0, 1));
        }

        [Test]
        public async Task Foo()
        {
            Assert.IsNotNull(Guid.NewGuid().ToString("N"));

            await Task.Delay(new TimeSpan(0, 0, 1));
        }
    }
````

**Comment by CharliePoole** [comment]
Does the test run correctl under the nunit console or gui runners?

**Comment by ErikCL** [comment]
Yes it runs correctly under the current stable release of the offical GUI test runner 2.6.4 (*win*) found at http://www.nunit.org/index.php?p=download

**Comment by CharliePoole** [comment]
Thanks @ErikCL That will help us and it's good news! If the error were in NUnit 2.6.4, which we are not planning to maintain further, you'd be stuck. Since it only occurs in the adapter, it's should be something we can fix!

Please add info about the environment under which you are running the adapter.

**Comment by ErikCL** [comment]
I've noticed this bug on my local environment with Windows 8.1 running Visual Studio Ultimate 2013 Update 5. I've also noticed this bug on the standard Visual Studio Online hosted build server.

**Comment by OsirisTerje** [comment]
Confirmed.  [Repro here](https://github.com/OsirisTerje/NUnit.Issues). folder Issue87

**Comment by rprouse** [comment]
Fixed by #95
